### PR TITLE
Adding programmatic descriptions

### PR DESCRIPTION
**Pull request by samshuster**
### Summary of Changes

Request as part of https://github.com/lyft/amundsen/issues/147

Blocker to amundsenfrontendlibrary and metadata changes here:
https://github.com/lyft/amundsenfrontendlibrary/pull/381
https://github.com/lyft/amundsenmetadatalibrary/pull/104

This adds programmatic_descriptions as part of table object.

### Tests

None

### Documentation

None

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes, including screenshots of any UI changes. 
- [x] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public python functions and the classes in the PR contain docstrings that explain what it does
- [x] PR passes all tests documented in the [developer guide](https://github.com/lyft/amundsenfrontendlibrary/blob/master/docs/developer_guide.md#testing)

### CODE REVIEW

**Comment by samshuster** [comment]
hey @jinhyukchang this MR blocks metadata (https://github.com/lyft/amundsenmetadatalibrary/pull/104) from being built.

Let me know if you need anymore information on this one!

File: amundsen_common/models/table.py [hunk 1]
```diff
@@ -106,6 +106,18 @@ def default_if_none(arg: Optional[bool]) -> bool:
     return arg or False
 
 
+@attr.s(auto_attribs=True, kw_only=True)
+class ProgrammaticDescription:
+    source_id: str
```
Comment (jinhyukchang) [inline amundsen_common/models/table.py]: Shall we name it `source` to unify with `databuilder`?

Comment (samshuster) [reply amundsen_common/models/table.py]: thanks!

**Comment by samshuster** [comment]
@jinhyukchang done!

**Pull request merged**